sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,241 @@ use crate::{
     IndividTrip, PersonID, PersonSpec, Scenario, ScenarioGenerator, SpawnTrip, TripEndpoint,
     TripMode,
 };
-use abstutil::{Parallelism, Timer};
+use abstutil::Timer;
 use geom::{Distance, Duration, Time};
-use map_model::{BuildingID, BuildingType, Intersection, Map, PathConstraints, PathRequest};
+use map_model::{BuildingID, BuildingType, Map, PathConstraints, PathRequest};
 use rand::seq::SliceRandom;
 use rand::Rng;
 use rand_xorshift::XorShiftRng;
 
 impl ScenarioGenerator {
     // Designed in https://github.com/dabreegster/abstreet/issues/154
     pub fn proletariat_robot(map: &Map, rng: &mut XorShiftRng, timer: &mut Timer) -> Scenario {
-        let mut residences: Vec<(BuildingID, XorShiftRng)> = Vec::new();
-        let mut workplaces: Vec<BuildingID> = Vec::new();
-        let mut num_residences = 0;
+        let mut residents: Vec<BuildingID> = Vec::new();
+        let mut workers: Vec<BuildingID> = Vec::new();
+
+        let mut num_bldg_residential = 0;
+        let mut num_bldg_commercial = 0;
+        let mut num_bldg_mixed_residential_commercial = 0;
         for b in map.all_buildings() {
             match b.bldg_type {
                 BuildingType::Residential(num_ppl) => {
+                    // TODO is it expected that sometimes num_ppl is 0?
+                    // TODO better estimate, maybe based on parking, buildling size, etc
+                    let num_ppl = if num_ppl > 0 {
+                        num_ppl
+                    } else {
+                        rng.gen_range(1, 5)
+                    };
+
+                    assert!(num_ppl > 0);
                     for _ in 0..num_ppl {
-                        residences.push((b.id, abstutil::fork_rng(rng)));
+                        residents.push(b.id);
                     }
-                    num_residences += 1;
+                    num_bldg_residential += 1;
                 }
                 BuildingType::ResidentialCommercial(num_ppl) => {
+                    assert!(num_ppl > 0);
                     for _ in 0..num_ppl {
-                        residences.push((b.id, abstutil::fork_rng(rng)));
+                        residents.push(b.id);
+
+                        // TODO: currently we just assume work capacity is the same as residential
+                        // capacity, which is surely not true. How can we better estimate?
+                        workers.push(b.id);
                     }
-                    workplaces.push(b.id);
-                    num_residences += 1;
+                    num_bldg_mixed_residential_commercial += 1;
                 }
                 BuildingType::Commercial => {
-                    workplaces.push(b.id);
+                    // TODO: currently we just assume some constant for jobs per building.
+                    // A better metrics might be available parking or building size
```

Comment:
I have some ideas for making this better - similar to how we compute residential capacity, but changing the enum to `BuildingType::Commercial(usize)` breaks the binary serialized scenarios.

Are there docs for regenerating all the scenarios?

Reply:
Woops, it should be in https://dabreegster.github.io/abstreet/dev/index.html#building-map-data but isn't -- add `--scenario` to `import.sh`

Reply:
If it's alright with you, I'd like to follow up with the building capacity computations in a subsequent PR. What's included in this PR is already strictly an improvement in my estimation.

Reply:
SGTM. Just a thought on "A better metrics might be available parking": we pretty never have good data on available parking, so this direction of inference could be tough. Maybe in office park type areas, there'll be parking lots mapped in OSM and reasonably clearly associated with some buildings, at best.